### PR TITLE
Make core workflow more general

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: Ashish Tiwari <ashishjaitiwari15112000@gmail.com>

**Description**

This PR makes core E2E workflow more general so that it doesn't need an adapter to run always

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/test_adaptersv2.yaml [hunk 1]
```diff
@@ -0,0 +1,258 @@
+# This workflow will be called/referenced in each of the adapters and runs in the context of adapter
+name: Adapter Integration - Core v2
```
Comment (leecalcote) [inline .github/workflows/test_adaptersv2.yaml:2]: Oh, we're versioning now. Fancy. 😉

File: .github/workflows/test_adaptersv2.yaml [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    secrets:
+      token:
+        required: true
+
+jobs:
+  TestAdapter:
+    strategy:
+      fail-fast: false
+      matrix:
+        k8s_version: ['v1.20.11', 'v1.21.5', 'v1.22.2','v1.23.9']
```
Comment (leecalcote) [inline .github/workflows/test_adaptersv2.yaml:57]: No, 1.24 or 1.25?

Comment (Revolyssup) [reply .github/workflows/test_adaptersv2.yaml:57]: For now, due to problems in other adapters. We can keep this to 1.23.

Comment (leecalcote) [reply .github/workflows/test_adaptersv2.yaml:57]: @Revolyssup the issue here is that minikube and docker-desktop are both already running v1.25. We need to get to 1.25 now. What's preventing us from moving forward?

**Pull request merged**